src/toolbox/prover.rs

```diff
@@ -205,6 +239,22 @@ impl<'a> IsSigmaProtocol for Prover<'a> {
 
             commitments.push(encoding);
         }
+
+        // Now that we know all the clauses, make sure at least one is full
+        let mut have_one = false;
```

Comment:
Correct, the current code only checks for "is at least one clause fully-proved?"  It would be pretty simple to add in further checks, we'd just need to define what we think are invalid configurations.  I think our previous conversation was "is it OK for the user to provide keys for partially-proved clauses?" which at the time we said might be possible.  For example, you might want to prove `((A = G^x) && (B = G^y)) || ((A = G^x) && (C = G^z))`, which could equate to the situation where you always need two signers on a company check, and one must be the CFO.

At the end of the day, though, any checks here are simply hand-holding the user, because if we get the math right, invalid configurations will never verify.  This code just runs checks in our honest prover to help the user figure out what they're doing wrong.